### PR TITLE
Support VOTable 1.4 (issue #9474)

### LINKED ISSUE #9474: Support VOTable 1.4
### Description

VOTable version 1.4 is now an official IVOA recommendation.  The main difference from version 1.3 is that a new element, TIMESYS, is supported.  That element allows defining of metadata for temporal coordinates much like COOSYS defines metadata for celestial coordinates.

This issue is to ensure that VOTable 1.4 can be parsed and written while retaining the TIMESYS information.  Further use of TIMESYS can be handled with issue #8724.

### Additional context

VOTable 1.4 specifcation is [here](http://www.ivoa.net/documents/VOTable/20191021/).

### PR DESCRIPTION
**Pull request by tomdonaldson**
### Description

This pull request is to support the reading and writing of VOTable version 1.4.

Fixes #9474

### CODE REVIEW

**Comment by pllim** [comment]
@bsipocz , this won't get reviewed until Monday but can we still keep that 4.0 milestone for now, please? 🙏

**Comment by pllim** [comment]
@eteq , weren't you very excited about TIMESYS at IVOA? FYI in case you want to review.

File: astropy/io/votable/tests/vo_test.py [hunk 1]
```diff
@@ -718,7 +718,8 @@ def table_from_scratch():
 
 def test_open_files():
     for filename in get_pkg_data_filenames('data', pattern='*.xml'):
-        if filename.endswith('custom_datatype.xml'):
+        if (filename.endswith('custom_datatype.xml') or
+            filename.endswith('timesys_errors.xml')):
```
Comment (pllim) [inline astropy/io/votable/tests/vo_test.py]: Usually linter doesn't like the continuation of conditional at the same level as the nested statement.
```suggestion
                filename.endswith('timesys_errors.xml')):
```

File: astropy/io/votable/tree.py [hunk 2]
```diff
@@ -1724,6 +1724,123 @@ def epoch(self):
...
+        floating point literal; for convenience, the magic
+        strings "MJD-origin" (standing for 2400000.5) and
+        "JD-origin" (standing for 0) are also allowed.
+
+        The timeorigin attribute MUST be given unless the
+        time’s representation contains a year of a calendar
+        era, in which case it MUST NOT be present. In VOTables,
+        these representations currently are Gregorian calendar
+        years with xtype="timestamp", or years in the Julian
+        or Besselian calendar when a column has yr, a or Ba as
```
Comment (pllim) [inline astropy/io/votable/tree.py]: Oxford comma! 😸 
```suggestion
        or Besselian calendar when a column has yr, a, or Ba as
```

File: astropy/io/votable/tree.py [hunk 3]
```diff
@@ -1724,6 +1724,123 @@ def epoch(self):
...
+        its unit and no time origin is given.
+        """
+        return self._timeorigin
+
+    @timeorigin.setter
+    def timeorigin(self, timeorigin):
+        if (timeorigin is not None and
+            timeorigin != 'MJD-origin' and timeorigin != 'JD-origin'):
```
Comment (pllim) [inline astropy/io/votable/tree.py]:
```suggestion
                timeorigin != 'MJD-origin' and timeorigin != 'JD-origin'):
```

File: astropy/io/votable/tree.py [hunk 4]
```diff
@@ -3242,15 +3385,16 @@ def __init__(self, ID=None, id=None, config=None, pos=None, version="1.3"):
...
         self._infos = HomogeneousList(Info)
         self._resources = HomogeneousList(Resource)
         self._groups = HomogeneousList(Group)
 
         version = str(version)
-        if version not in ("1.0", "1.1", "1.2", "1.3"):
+        if version not in ("1.0", "1.1", "1.2", "1.3", "1.4"):
             raise ValueError("'version' should be one of '1.0', '1.1', "
-                             "'1.2', or '1.3'")
+                             "'1.2', '1.3' or '1.4'")
```
Comment (pllim) [inline astropy/io/votable/tree.py]:
```suggestion
                             "'1.2', '1.3', or '1.4'")
```

File: astropy/io/votable/tree.py [hunk 5]
```diff
@@ -3268,10 +3412,10 @@ def version(self):
     @version.setter
     def version(self, version):
         version = str(version)
-        if version not in ('1.1', '1.2', '1.3'):
+        if version not in ('1.1', '1.2', '1.3', '1.4'):
             raise ValueError(
                 "astropy.io.votable only supports VOTable versions "
-                "1.1, 1.2 and 1.3")
+                "1.1, 1.2, 1.3 and 1.4")
```
Comment (pllim) [inline astropy/io/votable/tree.py]:
```suggestion
                "1.1, 1.2, 1.3, and 1.4")
```

File: docs/io/votable/references.txt [hunk 6]
```diff
@@ -1,6 +1,6 @@
 .. _BINARY: http://www.ivoa.net/documents/PR/VOTable/VOTable-20040322.html#ToC27
```
Comment (gpdf) [inline docs/io/votable/references.txt]: Does it make sense to change this and all the other references to IVOA PRs in this section (i.e., also BINARY2 and TABLEDATA) to be to the appropriate RECs?  (It's not on the main topic of this pull request, of course; I could do a separate pull request on that after this one is merged.)

**Comment by gpdf** [comment]
It would be cool if you could pull in #9470 as well.

File: docs/io/votable/references.txt (hunk 6 above)
Comment (tomdonaldson) [reply docs/io/votable/references.txt]: Yes, I agree, but was leery of scope creep and rushing these in without at least sanity checking that the references are still appropriate.  A new PR makes sense to me.  Thanks for taking a look at this PR!

File: astropy/io/votable/tree.py (hunk 5 above)
Comment (tomdonaldson) [reply astropy/io/votable/tree.py]: Thanks for the review.  I've implemented all of the above suggestions.

**Comment by msdemlei** [comment]
I've not really looked sufficiently diligently to call this a review (and I'm also not familiar enough with the inner workings of astropy.votable to make me a good reviewer), but with these caveats all looks fine to me.

Of course, this is just the basis on which code that actually does something useful can be built.  As part of the design process, I wrote some proof-of-concept code that might give some inspiration: https://volute.g-vo.org/svn/trunk/projects/votable/timesys-poc/

**Comment by pllim** [comment]
Great, let's get this in then. Thanks for the work and reviews, everyone!

**Pull request merged**